File: intro/week6/psets6/bleep/bleep.py
from cs50 import get_string
from sys import argv


def main():

    if not len(argv) == 2:
        print("Usage: python bleep.py dictionary")
        exit(1)

    file = open("D:/CS50/week6/psets6/bleep/banned.txt", 'r')
    bannedWords = set()
    for ch in file:
        bannedWords.add(ch.strip('\n'))
    
    message = get_string("What message would you like to censor?\n")
    
    words = message.split()

    # for word in words:
    #     star = []
    #     for bannedWord in bannedWords:
    #         if bannedWord == word.lower():
    #             words.remove(word)
    #             for _ in range(len(word)):
    #                 star.append('*')
    #             starStr = "".join(star)
    #             words.append(starStr)
    for word in words:
        star = ""
        for bannedWord in bannedWords:
            if bannedWord == word.lower():
                words.remove(word)
                for ch in word:
                    star += "*"
                words.append(star)
                    

    for word in words:
        print(word, end=" ")

    file.close()

if __name__ == "__main__":
    main()

File: intro/week6/psets6/bleep/bleep.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(int argc, char *argv[])
{
    // if (argc != 2)
    // {
    //     printf("Usage: ./bleep.c dictionary");
    //     return 1;
    // }   

    FILE *f = fopen("D:/CS50/week6/psets6/bleep/banned.txt", "r");
    if (f == NULL)
    {
        printf("File can not be opened\n");
        return 2;
    }

    char buffer[555];
    char *bannedWords[8];
    for(int t = 0; t < 8; t++)
    {
        bannedWords[t] = malloc(50);
    }
    int i = 0;
    while(fgets(buffer, 50, f))
    {
        strcpy(bannedWords[i], buffer);
        i++;
    }

    for(int s = 0; s <= i; s++)
        printf("%s", bannedWords[s]);
    
    fclose(f);

    return 0;
}